### ISSUE TITLE
Compact/packed format

### ISSUE THREAD

**Issue by arthurprs** [pyfisch/cbor#12]
Hi, it'd be nice to use (have an option to use) a compact/packed serialization format (like bincode and rmp) that serializes structs like lists (thus relying on the order of the fields). This allows faster (de)serialization and more compact output.

rmp (msgpack)

```
[
  2837513946597, 
  123456, 
  [
    1, 
    []
  ], 
  [
    [
      2, 
      []
    ], 
    200, 
    503, 
    520, 
    [
      1, 
      []
    ], 
    "text/html", 
    "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/33.0.1750.146 Safari/537.36", 
    "https://www.cloudflare.com/", 
    "/cdn-cgi/trace"
  ], 
  [
    "1.2.3.4", 
    8000, 
    "www.example.com", 
    [
      2, 
      []
    ]
  ], 
  [
    238, 
    []
  ], 
  [
    3, 
    []
  ], 
  "192.168.1.1", 
  "metal.cloudflare.com", 
  "10.1.2.3", 
  123456, 
  "10c73629cce30078-LAX"
]
```

cbor

```
{
    "timestamp": 2837513946597,
    "zone_id": 123456,
    "zone_plan": null,
    "http": {
        "protocol": null,
        "status": 200,
        "host_status": 503,
        "up_status": 520,
        "method": null,
        "content_type": "text/html",
        "user_agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/33.0.1750.146 Safari/537.36",
        "referer": "https://www.cloudflare.com/",
        "request_uri": "/cdn-cgi/trace"
    },
    "origin": {
        "ip": "1.2.3.4",
        "port": 8000,
        "hostname": "www.example.com",
        "protocol": null
    },
    "country": null,
    "cache_status": null,
    "server_ip": "192.168.1.1",
    "server_name": "metal.cloudflare.com",
    "remote_ip": "10.1.2.3",
    "bytes_dlv": 123456,
    "ray_id": "10c73629cce30078-LAX"
}
```

**Comment by pyfisch** [comment]
I am open to add this option. But the CBOR spec mentions in the last paragraph of Section 3.7 that constrained devices should use integer keys for maps. The first 24 keys of a struct would only take one byte each. I prefer this solution because it preserves more semantics and is easier to extend. Would you be okay with using integer keys if an option is used?

**Comment by arthurprs** [comment]
I think it's a worthy addition.

**Comment by pyfisch** [comment]
@arthurprs It has been a long time but finally packed encoding and decoding has landed in version 0.5.1. It would be great if you'd test the new encoding and tell me if it fits your needs or needs more improvement.
